core/src/main/java/org/mule/runtime/core/internal/registry/TypeBasedTransformerResolver.java

```diff
@@ -93,6 +94,32 @@ public Transformer resolve(DataType source, DataType result) throws ResolverExce
     }
 
     transformer = getNearestTransformerMatch(trans, source.getType(), result.getType());
+    // If an exact mach is not found, we have a 'second pass' transformer that can be used to converting to String or
+    // byte[]
+    Transformer secondPass;
+    if (transformer == null) {
+      // If no transformers were found but the outputType type is String or byte[] we can perform a more general search
+      // using Object.class and then convert to String or byte[] using the second pass transformer
+      if (result.getType().equals(String.class)) {
+        secondPass = objectToString;
+      } else if (result.getType().equals(byte[].class)) {
+        secondPass = objectToByteArray;
+      } else {
+        return null;
+      }
+      // Perform a more general search
+      trans = registry.lookupTransformers(source, DataType.OBJECT);
+
+      transformer = getNearestTransformerMatch(trans, source.getType(), result.getType());
+      if (transformer != null) {
+        transformer = new TransformerChain(transformer, secondPass);
+        try {
+          registry.registerTransformer(transformer);
+        } catch (MuleException e) {
+          throw new ResolverException(e.getI18nMessage(), e);
```

Comment:
unreated to this change, but ResolverException should:
* not be API
* extend TransformerException or be mapped to `TRANSFORMATION` errorType.

Create a jira for this